core/code/entity_decode.js

```diff
@@ -60,10 +60,21 @@ function parseArtifactDetail(arr) {
   };
 }
 
+function parseHistoryDetail(bitarray, required) {
+  if (bitarray == null && !required) return null;
+  if (bitarray == null) bitarray = 0;
```

Comment:
This is handled at call

```suggestion
function parseHistoryDetail(bitarray) {
```